File: app/src/main/java/eu/warble/pjapp/ui/splash/SplashActivity.java
package eu.warble.pjapp.ui.splash;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.v4.content.ContextCompat;

import eu.warble.pjapp.R;
import eu.warble.pjapp.ui.MainActivity;
import eu.warble.pjapp.ui.base.BaseActivity;
import eu.warble.pjapp.ui.login.LoginActivity;

public class SplashActivity extends BaseActivity<SplashPresenter> {

    @Override
    protected SplashPresenter createPresenter() {
        return new SplashPresenter(this);
    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        setTheme(R.style.SplashTheme);
    }

    public void showMessage(String message){
        Snackbar.make(findViewById(android.R.id.content), message, Snackbar.LENGTH_LONG)
                .show();
    }

    public void showConnectionError(){
        Snackbar.make(findViewById(android.R.id.content), getString(R.string.connect_error), Snackbar.LENGTH_INDEFINITE)
                .setAction(R.string.retry, l -> presenter.loadData())
                .setActionTextColor(ContextCompat.getColor(SplashActivity.this, R.color.colorAccent))
                .show();
    }

    public void startLoginActivity(){
        Intent intent = new Intent(SplashActivity.this, LoginActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
        startActivity(intent);
        finish();
    }

    public void startMainActivity(){
        Intent intent = new Intent(SplashActivity.this, MainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
        startActivity(intent);
        finish();
    }

    public void startMainActivityWithNoInternetMode() {
        Intent intent = new Intent(SplashActivity.this, MainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
        intent.putExtra("internet", false);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/eu/warble/pjappkotlin/mvp/splash/SplashActivity.kt
package eu.warble.pjappkotlin.mvp.splash

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v4.content.ContextCompat
import eu.warble.pjappkotlin.R
import eu.warble.pjappkotlin.mvp.ApplicationNavigator
import eu.warble.pjappkotlin.mvp.BaseActivity
import eu.warble.pjappkotlin.mvp.login.LoginActivity
import eu.warble.pjappkotlin.mvp.main.MainActivity
import eu.warble.pjappkotlin.mvp.mainguest.MainGuestActivity
import eu.warble.pjappkotlin.utils.Injection

class SplashActivity : BaseActivity(), SplashContract.View {
    override lateinit var presenter: SplashContract.Presenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTheme(R.style.SplashTheme)
        initPresenter()
    }

    private fun initPresenter() {
        presenter = SplashPresenter(
                Injection.provideStudentDataRepository(applicationContext),
                this,
                applicationContext
        )
    }

    override fun goToLoginActivity() {
        val intent = Intent(this, LoginActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK
        startActivity(intent)
        overridePendingTransition(0, 0)
    }

    override fun goToMainActivity() {
        val intent = Intent(this, MainActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK
        startActivity(intent)
        overridePendingTransition(0, 0)
    }

    override fun goToMainActivityWithNoInternetMode() {
        val intent = Intent(this, MainActivity::class.java).apply {
            flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK
            putExtra("internet", false)
        }
        startActivity(intent)
        overridePendingTransition(0, 0)
    }

    override fun onResume() {
        super.onResume()
        presenter.start()
    }

    override fun showConnectionError() {
        Snackbar.make(findViewById(android.R.id.content), getString(R.string.connect_error), Snackbar.LENGTH_INDEFINITE)
                .setAction(R.string.retry) { presenter.checkUserLogIn() }
                .setActionTextColor(ContextCompat.getColor(this@SplashActivity, R.color.colorAccent))
                .show()
    }

}
